### PR TITLE
app key renaming

### PR DESCRIPTION
**Pull request by spartan218**
These are just the changes to areas where we use account ID
for authorization to be renamed.

All Tests in pre-commit.sh pass.

### CODE REVIEW

**Comment by spartan218** [comment]
I think I disagree with coverage.
The area where I think it see less coverage is in 
`raw_api.py` because I changed the account ID we compare to from the response
rather than the account ID from the `b2_authorize_account` call which I believe is 
the proper way to check an account ID.

File: b2sdk/api.py [hunk 1]
```diff
@@ -127,37 +127,37 @@ def authorize_automatically(self):
...
-        :type account_id_or_key_id: str
+        :param application_key_id: application key ID
+        :type application_key_id: str
         :param application_key: user's application key
         :type application_key: str
         """
         # Clean up any previous account info if it was for a different account.
         try:
-            old_account_id = self.account_info.get_account_id()
+            old_id = self.account_info.get_application_key_id()
```
Comment (spartan218) [inline b2sdk/api.py]: Woops this was supposed to be in a different PR

Comment (spartan218) [reply b2sdk/api.py]: Let me know if you feel strongly about this in a different PR and I can remove it and maybe add a test to make sure the cache is cleared?

File: b2sdk/account_info/sqlite_account_info.py [hunk 2]
```diff
@@ -320,9 +320,9 @@ def get_account_id(self):
         """
         return self._get_account_info_or_raise('account_id')
 
-    def get_account_id_or_app_key_id(self):
+    def get_application_key_id(self):
         """
-        Return account ID or appliction key ID.
+        Return application key ID or Account ID.
```
Comment (ppolewicz) [inline b2sdk/account_info/sqlite_account_info.py]:
```suggestion
        Return application key id.
```
How `application key id` and `account id` relate to each other, why they can be equal and when, is outside of scope of this docstring. Whoever reads this, should already know what application key id is, or should go check it in some other docs.

File: README.md [hunk 3]
```diff
@@ -69,9 +76,9 @@ Before checking in, use the `pre-commit.sh` script to check code formatting, run
 unit tests, run integration tests etc.
 
 The integration tests need a file in your home directory called `.b2_auth`
-that contains two lines with nothing on them but your account ID and application key:
+that contains two lines with nothing on them but your application Key ID and application key:
```
Comment (ppolewicz) [inline README.md]: please decide on the spelling. As of now in this PR we can see it as
* application key id
* application Key ID
* application key Id

I personally prefer the first one, but the most important bit is to be consistent (so that we can one day grep the repo to find it etc)

File: b2sdk/account_info/in_memory.py [hunk 4]
```diff
@@ -133,13 +133,13 @@ def get_account_id(self):
...
-    def get_account_id_or_app_key_id(self):
+    def get_application_key_id(self):
         """ 
-        Returns the account id or key id used to authenticate
+        Returns the application key id used to authenticate
 
         :rtype: str
         """
-        return self._account_id_or_app_key_id
+        return self._app_key_id
```
Comment (ppolewicz) [inline b2sdk/account_info/in_memory.py]: The variable containing it is sometimes called:
* application_key_id
* _app_key_id / app_key_id

I think there is no need to shorten it (we all use variable completion in our IDEs anyway), but there is a great benefit of having it consistently named across the entire library. When the argument is renamed in each layer, it becomes difficult to read the code.

File: b2sdk/raw_api.py [hunk 5]
```diff
@@ -548,9 +548,9 @@ def test_raw_api_helper(raw_api):
     this test will break and we'll have to do something about
     it.
     """
-    account_id = os.environ.get('TEST_ACCOUNT_ID')
-    if account_id is None:
-        print('TEST_ACCOUNT_ID is not set.', file=sys.stderr)
+    application_key_id = os.environ.get('TEST_APPLICATION_KEY_ID')
```
Comment (ppolewicz) [inline b2sdk/raw_api.py]: this requires changes in the build environment, please make sure you ask Brian Beach to adjust the CI configuration to provide the key on both env variables during the transition period

**Comment by ppolewicz** [comment]
Also, the changelog entry is missing. Please note that you've changed signature of AbstractAccountInfo (and all children) from `.get_account_id_or_app_key_id()` to `.get_application_key_id()`, and also the argumnt name was changed in `_set_auth_data`. While for the first change we could simply provide a proxy method with a deprecation warning, the second one is tougher and would require some crafty decorator (which doesn't mean it's impossible, it's just less clean).

Therefore a decision needs to be made, whether we want to bump the major version because of this change, or not.

File: b2sdk/account_info/sqlite_account_info.py (hunk 2 above)
Comment (spartan218) [reply b2sdk/account_info/sqlite_account_info.py]: Are you suggesting that I leave the original docstring or that its fine?

File: b2sdk/raw_api.py (hunk 5 above)
Comment (spartan218) [reply b2sdk/raw_api.py]: I thought that the CI configuration only supplied to strings which it mapped to environment variable names. I will check with Brian ty.

File: b2sdk/account_info/sqlite_account_info.py (hunk 2 above)
Comment (ppolewicz) [reply b2sdk/account_info/sqlite_account_info.py]: The old version was:
```
Return account ID or appliction key ID.
```
Your change is:
```
Return application key ID or Account ID.
```
and I suggest to:
```
Return application key id.
```
(if you want to accept my suggestion, you can click a button on github)

File: b2sdk/raw_api.py (hunk 5 above)
Comment (ppolewicz) [reply b2sdk/raw_api.py]: Yes. For the transition period we should export this as **both** `TEST_ACCOUNT_ID` and `TEST_APPLICATION_KEY_ID`.

File: b2sdk/account_info/sqlite_account_info.py (hunk 2 above)
Comment (spartan218) [reply b2sdk/account_info/sqlite_account_info.py]: I believe after our convention discussion in slack it will be 
`Return application key ID.`

File: b2sdk/account_info/in_memory.py [hunk 6]
```diff
@@ -35,8 +35,8 @@ class InMemoryAccountInfo(UrlPoolAccountInfo):
     """
     Holder for all account-related information that needs to be kept
     between API calls, and between invocations of the command-line
-    tool.  This includes: account id, application key, auth tokens,
-    API URL, download URL, and uploads URLs.
+    tool.  This includes: account id, application key ID, application key,
```
Comment (ppolewicz) [inline b2sdk/account_info/in_memory.py]:
```suggestion
    tool.  This includes: account ID, application key ID, application key,
```

File: test/test_account_info.py [hunk 7]
```diff
@@ -280,7 +280,7 @@ def test_upgrade_1_default_allowed(self):
 
     def test_upgrade_2_default_app_key(self):
         """
-        The 'account_id_or_app_key_id' field should default to the account id.
+        The 'application_key_id' field should default to the account id.
```
Comment (ppolewicz) [inline test/test_account_info.py]:
```suggestion
        The 'application_key_id' field should default to the account ID.
```

File: b2sdk/raw_api.py (hunk 5 above)
Comment (spartan218) [reply b2sdk/raw_api.py]: @bwbeach should I make a check for either `TEST_ACCOUNT_ID` or `TEST_APPLICATION_KEY_ID`?

**Comment by ppolewicz** [comment]
Coveralls has a false positive here due to `raw_api` not being covered by unit tests (we use the simulator in our unit tests). Actually `raw_api` is being tested by travis, but we don't tell coveralls about it (whether it was a deliberate decision to handle it this way or not, I think it might be a good decision, because those tests don't really do much and I'd rather keep coverage tracking meaningful for `b2http`).

**Comment by ppolewicz** [comment]
continued in #28
